### PR TITLE
fix: Clicking twice to announce makes 2 posts - MEED-1784

### PR DESCRIPTION
**Pull request by SaraBoutej**
Make the announce button disabled when clicking for the first time to announce a challenge.

### CODE REVIEW

File: portlets/src/main/webapp/vue-app/engagement-center/components/rules/RuleDetailDrawer.vue [hunk 1]
```diff
@@ -141,6 +141,7 @@ export default {
     MAX_LENGTH: 1300,
     editorFocus: false,
     editor: false,
+    canAnnounce: false
```
Comment (boubaker) [inline portlets/src/main/webapp/vue-app/engagement-center/components/rules/RuleDetailDrawer.vue]:
```suggestion
    sending: false
```

File: portlets/src/main/webapp/vue-app/engagement-center/components/rules/RuleDetailDrawer.vue [hunk 2]
```diff
@@ -199,9 +200,7 @@ export default {
       return this.program?.audienceId;
     },
     announceDisabled() {
-      return !this.validInput
-          || !this.comment
-          || !this.comment.length;
+      return !this.validInput || !this.comment || !this.comment.length || this.canAnnounce;
```
Comment (boubaker) [inline portlets/src/main/webapp/vue-app/engagement-center/components/rules/RuleDetailDrawer.vue]: Loading effect is sufficient to disable the button in fact. The `disable` value should be different from `loading` value of button.

```suggestion
      return !this.validInput || !this.comment || !this.comment.length;
```

please change the button definition to use `v-btn` instead with current definition : 
```
<v-btn
  :loading="sending"
  :disabled="buttonDisabled"<!-- to rename variable IMO to be more consistent -->
```

File: portlets/src/main/webapp/vue-app/engagement-center/components/rules/RuleDetailDrawer.vue [hunk 3]
```diff
@@ -267,6 +266,7 @@ export default {
       };
 
       this.$refs.ruleDetailDrawer.startLoading();
+      this.canAnnounce = true;
```
Comment (boubaker) [inline portlets/src/main/webapp/vue-app/engagement-center/components/rules/RuleDetailDrawer.vue]: No need to drawer loading to change here. In fact, it should be made inside a watch this way : 
```
watch: {
  sending() {
    if (this.sending) {
      this.$refs.ruleDetailDrawer.startLoading();
    } else {
      this.$refs.ruleDetailDrawer.endLoading();
    }
  }
}
```

```suggestion
      this.canAnnounce = true;
```

File: portlets/src/main/webapp/vue-app/engagement-center/components/rules/RuleDetailDrawer.vue [hunk 4]
```diff
@@ -287,7 +287,10 @@ export default {
           }
           this.$engagementCenterUtils.displayAlert(msg, 'error');
         })
-        .finally(() => this.$refs.ruleDetailDrawer.endLoading());
+        .finally(() => {
+          this.$refs.ruleDetailDrawer.endLoading();
+          this.canAnnounce = false;
+        });
```
Comment (boubaker) [inline portlets/src/main/webapp/vue-app/engagement-center/components/rules/RuleDetailDrawer.vue]: same as previous remark in fact

```suggestion
        .finally(() => this.sending = false);
```

File: portlets/src/main/webapp/vue-app/engagement-center/components/rules/RuleDetailDrawer.vue (hunk 1 above)
Comment (SaraBoutej) [reply portlets/src/main/webapp/vue-app/engagement-center/components/rules/RuleDetailDrawer.vue]: done

File: portlets/src/main/webapp/vue-app/engagement-center/components/rules/RuleDetailDrawer.vue (hunk 2 above)
Comment (SaraBoutej) [reply portlets/src/main/webapp/vue-app/engagement-center/components/rules/RuleDetailDrawer.vue]: done

File: portlets/src/main/webapp/vue-app/engagement-center/components/rules/RuleDetailDrawer.vue (hunk 3 above)
Comment (SaraBoutej) [reply portlets/src/main/webapp/vue-app/engagement-center/components/rules/RuleDetailDrawer.vue]: done

File: portlets/src/main/webapp/vue-app/engagement-center/components/rules/RuleDetailDrawer.vue (hunk 4 above)
Comment (SaraBoutej) [reply portlets/src/main/webapp/vue-app/engagement-center/components/rules/RuleDetailDrawer.vue]: done

File: portlets/src/main/webapp/vue-app/engagement-center/components/rules/RuleDetailDrawer.vue [hunk 5]
```diff
@@ -90,17 +90,18 @@
...
           @click="close">
           {{ $t('rule.detail.label.cancel') }}
-        </button>
-        <button
-          :disabled="announceDisabled"
+        </v-btn>
+        <v-btn
+          :loading="sending"
+          :disabled="btnDisabled"
           class="ignore-vuetify-classes btn btn-primary"
```
Comment (boubaker) [inline portlets/src/main/webapp/vue-app/engagement-center/components/rules/RuleDetailDrawer.vue]:
```suggestion
          class="btn btn-primary"
```

File: portlets/src/main/webapp/vue-app/engagement-center/components/rules/RuleDetailDrawer.vue [hunk 6]
```diff
@@ -90,17 +90,18 @@
     <template v-if="!automaticRule && isActiveRule && editor" slot="footer">
       <div class="d-flex mr-2">
         <v-spacer />
-        <button
+        <v-btn
           class="ignore-vuetify-classes btn mx-1"
```
Comment (boubaker) [inline portlets/src/main/webapp/vue-app/engagement-center/components/rules/RuleDetailDrawer.vue]:
```suggestion
          class="btn me-2"
```

Comment (SaraBoutej) [reply portlets/src/main/webapp/vue-app/engagement-center/components/rules/RuleDetailDrawer.vue]: done

File: portlets/src/main/webapp/vue-app/engagement-center/components/rules/RuleDetailDrawer.vue (hunk 5 above)
Comment (SaraBoutej) [reply portlets/src/main/webapp/vue-app/engagement-center/components/rules/RuleDetailDrawer.vue]: done

**Review by boubaker — APPROVED** [review]

**Pull request merged**